src/Liquidity sawp Test.sol

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.19;
+
+import {Test, console} from "forge-std/Test.sol";
+import {LiquidityPool} from "../src/liquidityPoolSwapFunction.sol";
+import {MockERC20} from "../src/MockERC20.sol"; // Import the custom MockERC20
+
+contract SwapFunctionTest is Test {
+    LiquidityPool liquidityPool;
+    MockERC20 token0;
+    MockERC20 token1;
+    MockERC20 lpRewardsToken; // Add this line
+    address initialOwner = address(0x789);
+    address user1 = address(0x123);
+    address user2 = address(0x456);
+
+    // Constants for test
+    string constant POOL_NAME = "TestPool";
+    uint24 constant FEE = 3000; // Example fee value
+    uint160 constant INITIAL_SQRT_PRICE_X96 = 79228162514264337593543950336; // sqrt(2^96)
+    uint128 constant INITIAL_LIQUIDITY = 1000;
+    uint256 constant LOW_PRICE = 100;
+    uint256 HIGH_PRICE = 200;
+    uint256 constant TOKEN_AMOUNT = 1000 * 10**18;
+    uint256 constant SWAP_AMOUNT_IN = 500 * 10**18;
+    uint256 constant SWAP_AMOUNT_OUT_MIN = 1;
+
+      // Constants for test
+    string constant POOL_NAME = "TestPool";
+    uint24 constant FEE = 3000; // Example fee value
+    uint256 constant LOW_PRICE = 100;
+    uint256 constant HIGH_PRICE = 200;
+    uint256 constant TOKEN_AMOUNT = 1000 * 10**18;
+
```

Comment:
Hey dagim dont mint the tokens directly to the pool below. Make sure to Initiate the pool and add tokens from the user end as provider.  Also make sure you are using allPoolManager contract for the addition of Pool amounts